File: questions/1652. Defuse the Bomb/defuse.cpp
class Solution {
public:
    vector<int> decrypt(vector<int>& code, int k) {
        int n = code.size();
        vector<int> res(n);
        if(k == 0) return res;
        for(int i = 0; i < n; i++){
            if(k > 0){
                for(int s = i + 1; s < i + k + 1; s++){
                    res[i]+=code[s % n];
                }
            }else{
                for(int s = i - 1; s > i + k - 1; s--){
                    res[i] += code[(s + n)%n];
                }
            }
        }
        return res;
    }
};

File: questions/1652. Defuse the Bomb/defuse.py
class Solution:
    def decrypt(self, code: List[int], k: int) -> List[int]:
        n = len(code)
        res = [0]*n
        if k == 0: 
            return res
        for i in range(n):
            if k > 0:
                for s in range(i + 1, i + k + 1):
                    res[i] += code[s%n]
            else:
                for s in range(i - 1, i + k - 1, -1):
                    res[i] += code[(s + n)%n]
        return res
